src/NzbDrone.Core/Download/Clients/DownloadStation/Proxies/DiskStationProxyBase.cs

```diff
@@ -105,115 +179,109 @@ private void AuthenticateClient(DownloadStationSettings settings)
-                 { "query", "SYNO.API.Auth, SYNO.DownloadStation.Info, SYNO.DownloadStation.Task, SYNO.FileStation.List, SYNO.DSM.Info" },
+                 { "query", string.Join(",", apis.Keys) },
              };
 
-            var infoResponse = ProcessRequest<DiskStationApiInfoResponse>(DiskStationApi.Info, arguments, settings, "Get api version");
+            var requestBuilder = BuildRequest(settings, _apiInfo, HttpMethod.GET);
 
-            //TODO: Refactor this into more elegant code
-            var infoResponeDSAuth = infoResponse.Data["SYNO.API.Auth"];
-            var infoResponeDSInfo = infoResponse.Data["SYNO.DownloadStation.Info"];
-            var infoResponeDSTask = infoResponse.Data["SYNO.DownloadStation.Task"];
-            var infoResponseFSList = infoResponse.Data["SYNO.FileStation.List"];
-            var infoResponseDSMInfo = infoResponse.Data["SYNO.DSM.Info"];
+            foreach (var arg in arguments)
+            {
+                requestBuilder.AddQueryParam(arg.Key, arg.Value);
+            }
 
-            Resources[DiskStationApi.Auth] = infoResponeDSAuth.Path;
-            Resources[DiskStationApi.DownloadStationInfo] = infoResponeDSInfo.Path;
-            Resources[DiskStationApi.DownloadStationTask] = infoResponeDSTask.Path;
-            Resources[DiskStationApi.FileStationList] = infoResponseFSList.Path;
-            Resources[DiskStationApi.DSMInfo] = infoResponseDSMInfo.Path;
+            var infoResponse = ProcessRequest<DiskStationApiInfoResponse>(DiskStationApi.Info, requestBuilder.Build(), settings, "get api info");
 
-            switch (api)
+            foreach (var data in infoResponse.Data)
             {
-                case DiskStationApi.Auth:
-                    return Enumerable.Range(infoResponeDSAuth.MinVersion, infoResponeDSAuth.MaxVersion - infoResponeDSAuth.MinVersion + 1);
-                case DiskStationApi.DownloadStationInfo:
-                    return Enumerable.Range(infoResponeDSInfo.MinVersion, infoResponeDSInfo.MaxVersion - infoResponeDSInfo.MinVersion + 1);
-                case DiskStationApi.DownloadStationTask:
-                    return Enumerable.Range(infoResponeDSTask.MinVersion, infoResponeDSTask.MaxVersion - infoResponeDSTask.MinVersion + 1);
-                case DiskStationApi.FileStationList:
-                    return Enumerable.Range(infoResponseFSList.MinVersion, infoResponseFSList.MaxVersion - infoResponseFSList.MinVersion + 1);
-                case DiskStationApi.DSMInfo:
-                    return Enumerable.Range(infoResponseDSMInfo.MinVersion, infoResponseDSMInfo.MaxVersion - infoResponseDSMInfo.MinVersion + 1);
-                default:
-                    throw new DownloadClientException("Api not implemented");
+                if (apis.ContainsKey(data.Key))
+                {
+                    var key = GenerateInfoCacheKey(settings, apis[data.Key]);
+
+                    var info = _cacheInfo.Find(key);
```

Comment:
Don't check Find, just call Set. You'll want to refresh the data in the cache no matter what.